prime-router/src/main/kotlin/JurisdictionalFilters.kt

```diff
@@ -178,6 +178,34 @@ class AllowAll : JurisdictionalFilter {
 }
 
 
+
+/**
+ * Implements a quality check match.  If a row has valid data for all the columns, the row is selected.
+ * If any column name does not exist, nothing passes thru the filter.
+ * hasValidDataFor(columnName1, columnName2, columnName3, ...)
+ * If no columns are passed, all rows are selected.  So, any number of args is acceptable.
+ */
+class HasValidDataFor : JurisdictionalFilter {
+    override val name = "hasValidDataFor"
+
+    override fun getSelection(args: List<String>, table: Table, receiver: Receiver): Selection {
+        var selection = Selection.withRange(0, table.rowCount())
+
+        val columnNames = table.columnNames()
+        args.forEach { colName ->
+            if (columnNames.contains(colName)) {
+                val before = Selection.with(*selection.toArray())  // hack way to copy to a new Selection obj
+                selection = selection.andNot(table.stringColumn(colName).isEmptyString)
+                JurisdictionalFilters.logFiltering(before, selection,"$name($colName)", receiver)
```

Comment:
Check out the cool new detailed logging, via `logFiltering` which prints out the row number of every row that got the kibosh, and `logAllEliminated`, for cases where all the rows are eliminated, its silly to print them all out.